### PR TITLE
[VCDA-604] Convert all print statements in system tests to logger.debug statements

### PR DESCRIPTION
**Pull request by rocknes**
* Introduced a method in environment.py to provide a logger object.
* Replaced all print statements with logger.debug statements
* Redirected client request logs into another log file for sanity of log mesaages
* Fixed an issue with logger in client.py and environment,py where we were adding multiple handlers to the same logger.

Ran idisk_tests.py, catalog_tests.py and vapp_tests.py to verify that logging is working as desired.

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246)

### CODE REVIEW

File: pyvcloud/vcd/client.py [hunk 1]
```diff
@@ -508,20 +508,21 @@ def __init__(self,
...
-                                      '%(name)-15.15s | '
-                                      '%(module)-15.15s | '
-                                      '%(funcName)-12.12s | '
-                                      '%(message)s')
-        handler.setFormatter(formatter)
-        self._logger.addHandler(handler)
-
-        requests_logger = logging.getLogger("requests.packages.urllib3")
-        requests_logger.addHandler(handler)
-        requests_logger.setLevel(logging.DEBUG)
```
Comment (rocknes) [inline pyvcloud/vcd/client.py]: requests_logger is not used anywhere, it's dead code - so removing it.

**Comment by hodgesrm** [comment]
Reviewed 1 of 3 files at r1, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[system_tests/catalog_tests.py, line 41 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFIGuJ5_9KCX0pEvKu:-LDFIGuJ5_9KCX0pEvKv:bqmqaa8) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L41)):*
> ```Python
>         This test passes if the catalog is created successfully.
>         """
>         TestCatalog._logger = Environment.get_default_logger()
> ```

How do I set the default log level for tests?

---

*[system_tests/catalog_tests.py, line 48 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFJ1Tr-0tNssZ2-HM4:-LDFJ1TsAf1OmN8idDZk:bx0urar) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L48)):*
> ```Python
>         try:
>             catalog_resource = org.get_catalog(TestCatalog._test_catalog_name)
>             TestCatalog._logger.debug('Reusing test catalog.')
> ```

Overall I would recommend setting messages to INFO level and ensure that this level appears by default.  If the messages are going to a log it does not bother anyone and will always be available to help diagnose problems (instead of having to run tests again to get messages). 

---

*[system_tests/idisk_tests.py, line 338 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFIk_VBsla9uJNhacL:-LDFIk_VBsla9uJNhacM:b-irp8eh) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/idisk_tests.py#L338)):*
> ```Python
> 
>     def test_9999_cleanup(self):
>         """Release all resources help by this object for testing purposes."""
> ```

"help" should be "held"

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246)*

**Comment by rocknes** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[system_tests/catalog_tests.py, line 41 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFIGuJ5_9KCX0pEvKu:-LDFN1xKCPYnC1CB4RD8:bhre7w2) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L41)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

How do I set the default log level for tests?
</blockquote></details>

deafult log level is hard coded to DEBUG, for both client and system test logging.

---

*[system_tests/catalog_tests.py, line 48 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFJ1Tr-0tNssZ2-HM4:-LDFMOMA8HtxJQRqjSO9:b-vlct4b) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L48)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Overall I would recommend setting messages to INFO level and ensure that this level appears by default.  If the messages are going to a log it does not bother anyone and will always be available to help diagnose problems (instead of having to run tests again to get messages). 
</blockquote></details>

Right now the default log level is set to debug and we are logging everything using logger.debug statements.

For system tests, I have separated the request logging from other logging messages to clear up the clutter. Honestly speaking we don't log anything other that the requests in our pyvcloud code. The system tests are the exceptions. So, I would rather leave the logging level at DEBUG since nothing is being logged at INFO level currently.

---

*[system_tests/idisk_tests.py, line 338 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDFIk_VBsla9uJNhacL:-LDFMLv9Byu8ffFW3k12:bg9qigj) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/idisk_tests.py#L338)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

"help" should be "held"
</blockquote></details>

Oops, will correct the typo.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246)*

**Comment by sahithi** [comment]
Aritra, can you post few sample log lines from system tests and client?

---

Reviewed 1 of 3 files at r1, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[pyvcloud/system_test_framework/environment.py, line 242 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDI8YjxBcdjL4H6SGKn:-LDI8YjxBcdjL4H6SGKo:b-bdqlkq) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/pyvcloud/system_test_framework/environment.py#L242)):*
> ```Python
>             for pvdc_ref in pvdc_refs:
>                 if pvdc_ref.get('name').lower() == pvdc_name.lower():
>                     cls._logger.debug('Reusing existing ' + pvdc_name)
> ```

can we store cls._logger to a local variable? cls._logger.debug seems too long given the frequency of logger usage.

---

*[system_tests/catalog_tests.py, line 48 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDI8zGF9GFbIGMzprER:-LDI8zGF9GFbIGMzprES:bppsd4k) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L48)):*
> ```Python
>         try:
>             catalog_resource = org.get_catalog(TestCatalog._test_catalog_name)
>             TestCatalog._logger.debug('Reusing test catalog.')
> ```

same here. it would be better if it is as sample as
logger.debug('...')

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246#-:-LDIAsUi8DSpgQd-leCz:b-tiu2rz)*

**Comment by rocknes** [comment]
Sample log lines :

2018-05-23 15:24:32,343 | DEBUG | pyvcloud.system | environment     | instantiate_vapp               | Reusing existing vApp test_vapp.
2018-05-23 15:24:33,786 | DEBUG | pyvcloud.system | vapp_tests      | _create_empty_vapp             | Creating empty vApp.
2018-05-23 15:24:40,263 | DEBUG | pyvcloud.system | vapp_tests      | test_0000_setup                | Reusing customized vApp
2018-05-23 15:24:41,771 | DEBUG | pyvcloud.system | vapp_tests      | test_0020_add_delete_vm        | Adding vm additional-vm to vApp empty_vApp
2018-05-23 15:25:44,117 | DEBUG | pyvcloud.system | vapp_tests      | test_0020_add_delete_vm        | Removing vm additional-vm from vApp empty_vApp
2018-05-23 15:25:54,822 | DEBUG | pyvcloud.system | vapp_tests      | _power_on_vapp_if_possible     | Making sure vApp customized_vApp is powered on.

---

Reviewed 1 of 5 files at r3.
Review status: 3 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[pyvcloud/system_test_framework/environment.py, line 242 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDI8YjxBcdjL4H6SGKn:-LDNctjt1Jxwi1efRMZr:b-q595cy) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/pyvcloud/system_test_framework/environment.py#L242)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

can we store cls._logger to a local variable? cls._logger.debug seems too long given the frequency of logger usage.
</blockquote></details>

For environment.py cls._logger.debug isn't too long compared to logger.debug.
Using a local variable will however add the following line, 
logger = Environment.get_default_logger() to all the functions.
which I think is worse than the current situation.

---

*[system_tests/catalog_tests.py, line 48 at r3](https://reviewable.io/reviews/vmware/pyvcloud/246#-LDI8zGF9GFbIGMzprER:-LDNdEGB1oxW3FDF4OfT:blpf2px) ([raw file](https://github.com/vmware/pyvcloud/blob/42f907703b2058ae96fc45e76fecb27bdc8c05a5/system_tests/catalog_tests.py#L48)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

same here. it would be better if it is as sample as
logger.debug('...')
</blockquote></details>

Sure, will use local logger variable instead of using the class member _logger

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246#-:-LDNcVGXCz0C1T_Oyh8V:bphxeba)*

**Comment by sahithi** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/246#-:-LDOE_bZ7oo7-eLmYYz8:bnfp4nl)*

**Pull request merged**